Development/cpprest/http_utils.cpp

```diff
@@ -330,6 +330,98 @@ namespace web
+                        throw std::invalid_argument("invalid directive name, expected tchar");
+                    case directive_name:
+                        if (details::is_tchar(c)) { name.push_back(c); break; }
+                        result.push_back({ name, {} }); name.clear();
+                        if (U('=') == c) { state = directive_value; break; }
+                        if (U(';') == c) { state = pre_directive_name; break; }
+                        if (U(' ') == c || U('\t') == c) { state = pre_directive_value; break; }
+                        throw std::invalid_argument("invalid directive name, expected tchar");
+                    case pre_directive_value:
+                        if (U('=') == c) { state = directive_value; break; }
+                        if (U(' ') == c || U('\t') == c) { break; }
+                        if (U(';') == c) { state = pre_directive_name; break; }
+                        throw std::invalid_argument("invalid directive, expected '='");
+                    case directive_value:
+                        if (details::is_tchar(c)) { result.back().second.push_back(c); state = directive_value_token; break; }
+                        if (U('"') == c) { state = directive_value_quoted_string; break; }
+                        if (U(' ') == c || U('\t') == c) { break; }
+                        throw std::invalid_argument("invalid directive value, expected tchar or '\"'");
+                    case directive_value_token:
+                        if (details::is_tchar(c)) { result.back().second.push_back(c); break; }
+                        if (U(';') == c) { state = pre_directive_name; break; }
+                        if (U(' ') == c || U('\t') == c) { state = pre_directive; break; }
+                        throw std::invalid_argument("invalid directive value, expected tchar");
+                    case directive_value_quoted_string:
+                        if (U('"') == c) { state = pre_directive_name; break; }
+                        if (U('\\') == c) { state = directive_value_quoted_string_escape; break; }
+                        result.back().second.push_back(c);
+                        break;
+                    case directive_value_quoted_string_escape:
+                        result.back().second.push_back(c);
+                        state = directive_value_quoted_string;
+                        break;
+                    default:
+                        throw std::logic_error("unreachable code");
+                    }
+                }
+
+                if (!name.empty())
+                {
+                    switch (state)
+                    {
+                    case directive_name:
+                        result.push_back({ name, {} }); break;
+                    default:
+                        throw std::logic_error("unreachable code");
+                    }
+                }
+                return result;
```

Comment:
I think there are syntax errors that can reach this point, e.g. `foo=meow;bar="purr` (no closing `"`), which we should probably reject? This also applies to `parse_ptokens_header` of course...

Reply:
Resolved by #278.